src/optimizer/column_lifetime_analyzer.cpp

```diff
@@ -118,6 +124,8 @@ void ColumnLifetimeAnalyzer::VisitOperator(LogicalOperator &op) {
 		if (everything_referenced) {
 			break;
 		}
+		// first visit operator expressions to populate referenced columns
+		LogicalOperatorVisitor::VisitOperatorExpressions(op);
 		// filter, figure out which columns are not needed after the filter
 		column_binding_set_t unused_bindings;
 		ExtractUnusedColumnBindings(op.children[0]->GetColumnBindings(), unused_bindings);
```

Comment:
Same here - if we are explicitly calling `LogicalOperatorVisitor::VisitOperatorExpressions` we shouldn't need to call `StandardVisitOperator`